internal/characters/collei/collei.go

```diff
@@ -3,35 +3,52 @@ package collei
 import (
 	tmpl "github.com/genshinsim/gcsim/internal/template/character"
 	"github.com/genshinsim/gcsim/pkg/core"
+	"github.com/genshinsim/gcsim/pkg/core/event"
 	"github.com/genshinsim/gcsim/pkg/core/keys"
 	"github.com/genshinsim/gcsim/pkg/core/player/character"
 	"github.com/genshinsim/gcsim/pkg/core/player/character/profile"
 )
 
+var dendroEvents = []event.Event{event.OnOverload} // TODO: put all dendro events here
+
 func init() {
-	core.RegisterCharFunc(keys.Diona, NewChar)
+	core.RegisterCharFunc(keys.Collei, NewChar)
 }
 
 type char struct {
 	*tmpl.Character
+	burstExtendCount   int
+	sproutShouldExtend bool
+	sproutShouldProc   bool
+	sproutSrc          int
 }
 
-const normalHitNum = 4
-
 func NewChar(s *core.Core, w *character.CharWrapper, _ profile.CharacterProfile) error {
 	c := char{}
 	c.Character = tmpl.NewWithWrapper(s, w)
 
-	c.EnergyMax = 80
+	c.EnergyMax = 60
 	c.NormalHitNum = normalHitNum
-	c.BurstCon = 3
-	c.SkillCon = 5
+	c.BurstCon = 5
+	c.SkillCon = 3
+	c.burstExtendCount = 0
```

Comment:
in Go, you technically don't have to initialize these 4 ints and bools, they're 0 and false by default